src/main/java/com/rabbitmq/perf/Consumer.java

```diff
@@ -24,30 +24,34 @@
 import java.io.ByteArrayInputStream;
 import java.io.DataInputStream;
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
 public class Consumer extends ProducerConsumerBase implements Runnable {
 
-    private ConsumerImpl     q;
-    private final Channel          channel;
-    private final String           id;
-    private final String           queueName;
-    private final int              txSize;
-    private final boolean          autoAck;
-    private final int              multiAckEvery;
-    private final Stats stats;
-    private final int              msgLimit;
-    private final long             timeLimit;
-    private final CountDownLatch   latch = new CountDownLatch(1);
+    private ConsumerImpl                q;
+    private final Channel               channel;
+    private final String                id;
+    private final List<String>          queueNames;
+    private final int                   txSize;
+    private final boolean               autoAck;
+    private final int                   multiAckEvery;
+    private final Stats                 stats;
+    private final int                   msgLimit;
+    private final long                  timeLimit;
+    private final CountDownLatch        latch = new CountDownLatch(1);
+    private final Map<String, String>   ConsumerTagBranchMap = new HashMap<String, String>();
```

Comment:
Please use lower case for the first letter of properties. Use also a synchronized `Map`, with e.g. `Collections.synchronizedMap`, as it can be accessed from different threads.